src/TurbulenceClosures/turbulence_closure_implementations/CATKEVerticalDiffusivities/mixing_length.jl

```diff
@@ -145,30 +141,27 @@ end
 end
 
 @inline function convective_length_scaleᶜᶜᶜ(i, j, k, grid, closure, Cᶜ::Number, Cᵉ::Number, Cˢᵖ::Number,
-                                            velocities, tracers, buoyancy, surface_buoyancy_flux)
-
-    u = velocities.u
-    v = velocities.v
+                                            surface_buoyancy_flux, S², N², w★)
 
     Qᵇᵋ      = closure.minimum_convective_buoyancy_flux
     Qᵇ       = @inbounds surface_buoyancy_flux[i, j, 1]
-    w★       = turbulent_velocityᶜᶜᶜ(i, j, k, grid, closure, tracers.e)
-    w★²      = turbulent_velocityᶜᶜᶜ(i, j, k, grid, closure, tracers.e)^2
-    w★³      = turbulent_velocityᶜᶜᶜ(i, j, k, grid, closure, tracers.e)^3
-    S²       = shearᶜᶜᶜ(i, j, k, grid, u, v)
-    N²       = ℑzᵃᵃᶜ(i, j, k, grid, ∂z_b, buoyancy, tracers)
-    N²_above = ℑzᵃᵃᶜ(i, j, k+1, grid, ∂z_b, buoyancy, tracers)
+    w★¹      = @inbounds w★[i, j, k]
+    w★²      = @inbounds w★[i, j, k]^2
+    w★³      = @inbounds w★[i, j, k]^3
+    S²ᶜᶜᶜ    = ℑzᵃᵃᶜ(i, j, k, grid, S²)
+    N²_local = ℑzᵃᵃᶜ(i, j, k, grid, N²)
+    N²_above = ℑzᵃᵃᶜ(i, j, k+1, grid, N²)
 
     # "Convective length"
     # ℓᶜ ∼ boundary layer depth according to Deardorff scaling
     ℓᶜ = Cᶜ * w★³ / (Qᵇ + Qᵇᵋ)
     ℓᶜ = ifelse(isnan(ℓᶜ), zero(grid), ℓᶜ)
 
     # Figure out which mixing length applies
-    convecting = (Qᵇ > Qᵇᵋ) & (N² < 0)
+    convecting = (Qᵇ > Qᵇᵋ) & (N²_local < 0)
```

Comment:
"local" implies there is such a thing as a "non-local buoyancy gradient". But I don't think that's what you mean..